client/sites/settings/site/visibility/index.jsx

```diff
@@ -33,6 +33,42 @@ import { LaunchConfirmationModal } from './launch-confirmation-modal';
 import { LaunchSiteTrialUpsellNotice } from './launch-site-trial-notice';
 import './styles.scss';
 
+const createAgencyBillingMessage = ( agency, agencyLoading = false, agencyError = false ) => {
+	const agencyName = agency.name;
+	const existingWPCOMLicenseCount = agency.existing_wpcom_license_count || 0;
+	const price = formatCurrency( agency.prices?.actual_price, agency.prices?.currency );
```

Comment:
Bummer that the prices can still be `NaN` due to those potentially undefined arguments.

Did we intend to address that as part of that PR? 

I feel like if `actual_price` or `currency` is undefined, then the entire message is screwed as we won't really be able to say how much they will be charged.

Perhaps if the price doesn't exist, we can display the same less specific message that we display on loading / error?

Reply:
> Perhaps if the price doesn't exist, we can display the same less specific message that we display on loading / error?

That sounds good to me as well.

Reply:
Sure, I pushed an update that should address this feedback.